src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Calendar/Calendar.shared.cs

```diff
@@ -0,0 +1,533 @@
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateCalendarDays();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="SelectedDays"/> property.
+		/// </summary>
+		public static readonly BindableProperty SelectedDaysProperty =
+			BindableProperty.Create(nameof(SelectedDays), typeof(IReadOnlyCollection<DateTime>), typeof(Calendar), Enumerable.Empty<DateTime>().ToList(), propertyChanged: OnSelectedDaysChanged);
+
+		/// <summary>
+		/// Gets or sets selected days.
+		/// </summary>
+		public IReadOnlyCollection<DateTime> SelectedDays
+		{
+			get => (IReadOnlyCollection<DateTime>)GetValue(SelectedDaysProperty);
+			set => SetValue(SelectedDaysProperty, value);
+		}
+
+		static void OnSelectedDaysChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var daysToSelect = (IList<DateTime>)newValue;
+			var calendar = (Calendar)bindable;
+
+			if (calendar.SelectionMode is CalendarSelectionMode.SingleSelect
+				&& daysToSelect.Count > 1)
+			{
+				throw new InvalidOperationException(
+					$"Cannot select more than one day when working in {nameof(CalendarSelectionMode.SingleSelect)} mode");
+			}
+
+			calendar.SelectDays(daysToSelect);
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="Date"/> property.
+		/// </summary>
+		public static readonly BindableProperty DateProperty =
+			BindableProperty.Create(nameof(Date), typeof(DateTime), typeof(Calendar), default(DateTime), propertyChanged: OnDateChanged);
+
+		/// <summary>
+		/// Gets or sets the date.
+		/// </summary>
+		public DateTime Date
+		{
+			get => (DateTime)GetValue(DateProperty);
+			set => SetValue(DateProperty, value);
+		}
+
+		static void OnDateChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.UpdateCalendarDays();
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="SelectionMode"/> property.
+		/// </summary>
+		public static readonly BindableProperty SelectionModeProperty =
+			BindableProperty.Create(nameof(SelectionMode), typeof(CalendarSelectionMode), typeof(Calendar), CalendarSelectionMode.SingleSelect, propertyChanged: OnModeChanged);
+
+		/// <summary>
+		/// Gets or sets the mode, which determines how <see cref="Calendar"/> handles days selection.
+		/// </summary>
+		public CalendarSelectionMode SelectionMode
+		{
+			get => (CalendarSelectionMode)GetValue(SelectionModeProperty);
+			set => SetValue(SelectionModeProperty, value);
+		}
+
+		static void OnModeChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.SelectedDays = Enumerable.Empty<DateTime>().ToList();
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="Calendar"/> class.
+		/// </summary>
+		public Calendar()
+		{
+			var layoutRoot = new Grid
```

Comment:
@DamianAntonowicz here at XCT we have an [`UniformGridLayout`](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/UniformGrid.shared.cs) do you think that could be useful here? Since all items will have the same size

Reply:
I can see that UniformGrid is missing ColumnDefinitions and RowDefinitions - Calendar logic is based on those properties to show/hide days

Reply:
@DamianAntonowicz - Can you open a new enhancement Issue for adding `ColumnDefinitions` and `RowDefinitions` to `UniformGrid`? 

After `UniformGrid` has that functionality, we can implement here in `Calendar` in a later PR.

After that enhancement Issue is created, we can `Resolve` this conversation 👍

Reply:
ok, I will open enhancement

Reply:
@brminnick I've opened enhancement: https://github.com/xamarin/XamarinCommunityToolkit/issues/1370